### DISCUSSION TITLE
Portable version

### DISCUSSION THREAD

**Discussion by Z-Diezel** [baffler/Transparent-Twitch-Chat-Overlay#137] [Ideas]
Instead of of writing to AppData, can this be coded to be truly portable? I have multiple streamers that I game with, and I'd love to be able to create shortcuts with preset URLs in the custom CSS section like "https://www.twitch.tv/embed/(username)/chat?parent=twitch.tv"

(username) = the twitch streamer name

This is the CSS I made for a transparent overlay:

body { background-color: rgba(0,0,0,0) !important; font-size: 17px !important; text-shadow: -1px -1px 0 #000, 1px -1px 0 #000, -1px 1px 0 #000, 1px 1px 0 #000 !important; -webkit-mask-image: linear-gradient(to top, black 50%, transparent 100%); }
.user-notice-line { color: #fff !important; border-left: 0px !important; border-top: 2px solid #adadb8; padding-left: 2rem !important; }
user-notice-line--highlighted { color: #fff !important; border-left: 0px !important; border-top: 2px solid #adadb8; padding-left: 2rem !important; }
.link-fragment { color: #a476ea !important; }
.chat-author__display-name { color: #7e7e7e !important; }
.chat-line__message { color: #fff !important; font-weight: 500 !important; }
.chat-line__message--deleted-notice { color: #fff !important; }
.chat-line__moderation { color: #fff !important; }
/* Welcome message 
.CoreText-sc-cpl358-0 { visibility: hidden !important; } */
.chat-line__status { color: #fff !important; padding-top: 80% !important; }
.gqSSwW { color: #fff !important; }
.PGirV { display: flex !important; color: #fff !important; }
.cLxWNP { display: inline !important; color: #fff !important; }
.eEtqLn { display: inline !important; color: #fff !important; }
.dCOFmD { display: none !important; }
.gDuGLD  { display:none !important; }
.iFaqlo { display:none !important; }
.cJgrsK { background-color: rgba(0,0,0,0.2) !important; }
.mention-fragment { background-color: rgb(0,0,0,0.2) !important; color: #ffffff !important; }
.fwjUjn { display:none!important; }
.ciSpU { margin-bottom: -2px !important; }
.chat-input { display:none; }
.stream-chat .stream-chat-header { display:none; background-color: rgba(0,0,0,0) !important; color:white !important; }
.chat-room__notifcations { display:none; }
.tw-z-default { display:none; }
.tw-flex { background-color: rgba(0,0,0,0) !important; }
.tw-root { background-color: rgba(0,0,0,0) !important; }
.tw-root--theme-dark { background-color: rgba(0,0,0,0) !important; }
.stream-chat { background-color: rgba(0,0,0,0) !important; }
.chat-room { background-color: rgba(0,0,0,0) !important; }
.chat-list { background-color: rgba(0,0,0,0) !important; }
.scrollable-area { background-color: rgba(0,0,0,0) !important; color: white; }

**Comment by baffler** [comment]
Yeah, I will try to get this working for the next release

**Comment by Z-Diezel** [comment]
In addition to portability, being able to inject JS in the custom code as well as CSS would be wonderfully useful.

Here's my updated CSS for anyone that wants it:

body { background-color: rgba(0,0,0,0) !important; font-size: 16px !important; text-shadow: -1px -1px 0 #000, 1px -1px 0 #000, -1px 1px 0 #000, 1px 1px 0 #000 !important; -webkit-mask-image: linear-gradient(to top, black 60%, transparent 100%); margin-left: -12px !important; margin-right: -12px !important; }

/* Right side screen alignment (optional) */
body { text-align: right !important; }
.scrollable-area { margin-right: -16px !important; }
.simplebar-track.vertical { display: none !important; }

/* Raid */
.ekipGw {background-color: transparent !important; }
.fQBYwc { color: #fff !important; }

/* Hype Train popup */
.kAIqwe { display: none }

/* Shoutout */
.bTJhJp { color: #fff !important; }
.feJdGm { color: #fff !important; }

/* Redemption */
.bghctT { color: #fff !important; margin-right: 0px !important; }
.user-notice-line { color: #fff !important; border-left: 0px !important; border-top: 2px solid #a970ff; padding-left: 2rem !important; }
.user-notice-line--highlighted { color: #fff !important; border-left: 0px !important; border-top: 2px solid #a970ff; padding-left: 2rem !important; }
.bghctT { margin-right: 12px !important; }
.dEFOWq { padding-right: 0.5rem !important; margin-top: 0.3rem !important; margin-bottom: 0rem !important; }
.BcKcx { margin-top: 0rem !important; }

/* Clip bg */
.dFqqSQ { background-color: rgba(0,0,0,0.2) !important; }

/* Subscribe title color */
.kXZXTh { color: #815fc0 !important; }
.jSrrlW { color: #815fc0 !important; }
.jmrgPG { font-size: 16px !important; }
.WhLkF { font-size: 16px !important; display: inline !important; }

/* */
.link-fragment { color: #a476ea !important; text-decoration: none !important; }
.chat-author__display-name { color: #bf94ff !important; }
.chat-line__message { color: #fff !important; font-weight: 500 !important; }
.chat-line__message--emote { padding: 2px !important; }
.chat-line__message-body--highlighted { border: 0px solid #755ebc !important; background-color: rgba(117,94,188,0.4) !important; }
.chat-line__message--deleted-notice { color: #fff !important; }
.chat-line__moderation { color: #fff !important; }
.chat-line__message, .chat-line__moderation, .chat-line__status { padding: 0.4rem 2rem !important; }

/* Reply line */
.kAQTjE { color: #fff !important; }
.gEaRkX { color: #fff !important; }

/* Gift subs */
.CoreText-sc-1txzju1-0.cTWwJH { color: #fff !important; }
.CoreText-sc-1txzju1-0 cTWwJH { color: #fff !important; }
.hdNNzQ { fill: #fff !important; }
.cTWwJH { color: #fff !important; }
.hYMuUE { color: #815fc0 !important; }

/* Announcements chat section */
.announcement-line { border-left: 4px solid #9146ff !important; border-right: 0px !important; background-color: rgba(0,0,0,0.2) !important; }
.hxpcPU { background-color: rgba(0,0,0,0.2) !important; }
.kWErzy { background-color: rgba(0,0,0,0.2) !important; }

/* maybe unneeded */
.CoreText-sc-1txzju1-0 { padding-bottom: -6px !important; }

/* */
.gqSSwW { color: #fff !important; }
.PGirV { display: flex !important; color: #fff !important; }
.cLxWNP { display: inline !important; color: #fff !important; }
.eEtqLn { display: inline !important; color: #fff !important; }

/* Opt-out prompt */
.Layout-sc-1xcs6mc-0.fvoCQA.consent-banner { display:none !important; }

/* Chat header */
.jBYVfx { display:none !important; }
.dCOFmD { display: none !important; }
.gDuGLD  { display:none !important; }
.iFaqlo { display:none !important; }
.jaGgya { display:none !important; }
.cJgrsK { background-color: rgba(0,0,0,0.2) !important; }
.mention-fragment { background-color: rgb(0,0,0,0.2) !important; color: #ffffff !important; }

/* Chat Input Box Cluster */
.fwjUjn { display:none!important; }
.bGyiZe { display:none !important; }

/* Chat Input Box Formatting */
.chat-wysiwyg-input__editor { color: #fff !important; background-color: rgba(0,0,0,0) !important; font-size: 16px !important; text-shadow: -1px -1px 0 #000, 1px -1px 0 #000, -1px 1px 0 #000, 1px 1px 0 #000 !important; }
.chat-input__input-icons { display:none !important; }

/* Chat Redem Row */
.ejhEzS { display:none!important; }
.ciSpU { margin-bottom: -2px !important; }
.chat-input { display:none; }
.stream-chat .stream-chat-header { display:none; background-color: rgba(0,0,0,0) !important; color:white !important; }
.chat-room__notifcations { display:none; }
.tw-z-default { display:none; }
.tw-flex { background-color: rgba(0,0,0,0) !important; }
.tw-root { background-color: rgba(0,0,0,0) !important; }
.tw-root--theme-dark { background-color: rgba(0,0,0,0) !important; }
.stream-chat { background-color: rgba(0,0,0,0) !important; }
.chat-room { background-color: rgba(0,0,0,0) !important; }
.chat-list { background-color: rgba(0,0,0,0) !important; }
.scrollable-area { background-color: rgba(0,0,0,0) !important; color: white; }

/* Fading text */
.chat-line__message, .chat-line__message-highlight, .user-notice-line, .announcement-line, .CoreText-sc-1txzju1-0.gEaRkX { 
    animation: fadein 1s ease-in-out forwards,
               fadeout 12s ease-in-out 180s forwards; }
.CoreText-sc-1txzju1-0 { 
    animation: fadein 2s ease-in-out forwards,
               fadeout 3s ease-in-out 4s forwards; }
@keyframes fadein { 
    0% { opacity: 0; } 
    100% { opacity: 1; } }
@keyframes fadeout { 
    0% { opacity: 1; } 
    100% { opacity: 0; } }
